--- lib/msf/core/payload_generator.rb	2019-09-13 14:38:57.000000000 +0530
+++ lib/msf/core/payload_generatornew.rb	2019-09-30 10:21:02.326999820 +0530
@@ -32,6 +32,9 @@
     # @!attribute  arch
     #   @return [String] The CPU architecture to build the payload for
     attr_accessor :arch
+    # @!attribute  use_aapt2
+    #   @return [String] use aapt2 or not
+    attr_accessor :use_aapt2
     # @!attribute  badchars
     #   @return [String] The bad characters that can't be in the payload
     attr_accessor :badchars
@@ -385,7 +388,7 @@
       elsif payload.start_with? "android/" and not template.blank?
         cli_print "Using APK template: #{template}"
         apk_backdoor = ::Msf::Payload::Apk.new
-        raw_payload = apk_backdoor.backdoor_apk(template, generate_raw_payload)
+        raw_payload = apk_backdoor.backdoor_apk(template, generate_raw_payload, use_aapt2)
         gen_payload = raw_payload
       else
         raw_payload = generate_raw_payload
