--- lib/msf/core/payload_generator.rb.bak	2019-10-11 18:27:52.540999836 +0530
+++ lib/msf/core/payload_generator.rb	2019-10-11 18:36:10.565999890 +0530
@@ -32,6 +32,9 @@
     # @!attribute  arch
     #   @return [String] The CPU architecture to build the payload for
     attr_accessor :arch
+    # @!attribute  use_aapt2
+    #   @return [String] use aapt2 or not
+    attr_accessor :use_aapt2
     # @!attribute  badchars
     #   @return [String] The bad characters that can't be in the payload
     attr_accessor :badchars
@@ -130,6 +133,7 @@
       @encoder_space = opts.fetch(:encoder_space, @space)
 
       @framework  = opts.fetch(:framework)
+      @use_aapt2 = opts.fetch(:use_aapt2,false)
 
       raise ArgumentError, "Invalid Payload Selected" unless payload_is_valid?
       raise ::Msf::Simple::Buffer::BufferFormatError, "Invalid Format Selected" unless format_is_valid?
@@ -349,7 +353,7 @@
       elsif payload.start_with? "android/" and not template.blank?
         cli_print "Using APK template: #{template}"
         apk_backdoor = ::Msf::Payload::Apk.new
-        raw_payload = apk_backdoor.backdoor_apk(template, generate_raw_payload)
+        raw_payload = apk_backdoor.backdoor_apk(template, generate_raw_payload, use_aapt2)
         cli_print "Payload size: #{raw_payload.length} bytes"
         gen_payload = raw_payload
       else
